tapdb/assets_store.go

```diff
@@ -2640,6 +2677,20 @@ func (a *AssetStore) FetchAssetMetaByHash(ctx context.Context,
 	return assetMeta, nil
 }
 
+// WithCoinLock will acquire the coin lock and execute the passed function.
+// Callers to any coin related functions should use this to ensure that the
+// coin lock is held across the entire set of operations.
+func (a *AssetStore) WithCoinLock(fn func() error) error {
```

Comment:
See comments above about moving this. I think sort of unclear with it being so low level exactly how the caller should interact with it.